azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1Service.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.container.models;
+
+import com.google.common.base.Charsets;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.util.Yaml;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.text.StrSubstitutor;
+
+/**
+ * A class which creates the {@link V1Service} object based on the template and substituting values.
+ * Each instance of the class is unique to a template which will packages in the jar as a resource.
+ * After the instantiation, this class object can be reused to create different service objects.
+ */
+public class AzKubernetesV1Service {
+
+  private final String template;
+
+  /**
+   * Constructor for instantiation of this class
+   *
+   * @param templateName name of the resource within the package
+   * @throws IOException if the resource file is not readable
+   */
+  public AzKubernetesV1Service(String templateName) throws IOException {
```

Comment:
As the template is going to be constant for a release, it should be loaded only once and cached.

Reply:
The caller class can make sure to use a single instance everywhere. The intention is that we may have more than one template in the future.